### PR TITLE
DOC: Fix docstring cross-referencing

### PR DESCRIPTION
**Pull request by albertvillanova**
Links were not properly generated.

### CODE REVIEW

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -2007,8 +2007,8 @@ def compress(condition, a, axis=None, out=None):
     --------
     take, choose, diag, diagonal, select
     ndarray.compress : Equivalent method in ndarray
-    np.extract: Equivalent method when working on 1-D arrays
-    ufuncs-output-type
+    extract: Equivalent method when working on 1-D arrays
+    :ref:`ufuncs-output-type`
```
Comment (albertvillanova) [inline numpy/core/fromnumeric.py:2011]: I am not sure if the See Also section is the right place for this kind of link.

File: numpy/core/code_generators/ufunc_docstrings.py [hunk 2]
```diff
@@ -3596,7 +3596,7 @@ def add_newdoc(place, name, doc):
 
     See Also
     --------
-    lib.scimath.sqrt
+    numpy.lib.scimath.sqrt
```
Comment (rossbar) [inline numpy/core/code_generators/ufunc_docstrings.py:3599]: I think the problem here isn't that it's missing the `numpy.`, but that there is no `lib.scimath.sqrt` article being generated to link to. I've added an `autosummary` to the module docstring following the pattern set in the polynomial modules to fix this problem in 31fbcf5a3

**Review by rossbar — APPROVED** [review]
This takes a nice chunk out of #13114, from 764 -> 645 link warnings, thanks @albertvillanova !

Generally LGTM - I made a couple of minor tweaks including a proposed solution to an outstanding issue in 31fbcf5a3. LMK what you think @albertvillanova - if there are no objections I'll put this in soon.

**Comment by charris** [comment]
Thanks @albertvillanova and @rossbar .

File: numpy/core/code_generators/ufunc_docstrings.py (hunk 2 above)
Comment (albertvillanova) [reply numpy/core/code_generators/ufunc_docstrings.py]: Well seen!

**Comment by albertvillanova** [comment]
Thanks @rossbar for your thoroughly review.

**Pull request merged**